src/engine/LocalVocab.cpp

```diff
@@ -46,7 +46,7 @@ LocalVocabIndex LocalVocab::getIndexAndAddIfNotContained(std::string&& word) {
 
 // _____________________________________________________________________________
 const std::string& LocalVocab::getWord(LocalVocabIndex localVocabIndex) const {
-  if (localVocabIndex.get() > idsToWordsMap_.size()) {
+  if (localVocabIndex.get() >= idsToWordsMap_.size()) {
```

Comment:
Could you please add a unit test in retrospective that would have caught this bug in advance?